File: src/main/scala/jk_5/nailed/server/teamspeak/TeamspeakManager.scala
package jk_5.nailed.server.teamspeak

import java.util

import jk_5.nailed.server.NailedPlatform
import jk_5.nailed.server.teamspeak.api.JTS3ServerQuery
import org.apache.logging.log4j.LogManager

object TeamspeakManager {

  final val config = NailedPlatform.config.getConfig("teamspeak")
  final val enabled = config.getBoolean("enabled")
  final val host = config.getString("host")
  final val port = config.getInt("port")
  final val username = config.getString("username")
  final val password = config.getString("password")
  var connected = false
  private final val logger = LogManager.getLogger
  private final val server = new JTS3ServerQuery

  def start(){
    if(!enabled){
      logger.info("Teamspeak integration is disabled in the config")
      return
    }
    logger.info("Starting teamspeak integration")
    if(!this.server.connectTS3Query(this.host, this.port)){
      displayError()
    }
  }

  private def displayError(){
    val error: String = this.server.getLastError
    if (error != null) {
      System.out.println("Teamspeak error:")
      System.out.println(error)
      if (this.server.getLastErrorPermissionID != -1) {
        val permInfo: util.HashMap[String, String] = this.server.getPermissionInfo(this.server.getLastErrorPermissionID)
        if (permInfo != null) {
          System.out.println("Missing Permission: " + permInfo.get("permname"))
        }
      }
    }
  }
}

File: src/main/java/jk_5/nailed/server/teamspeak/TeamspeakManager.java
package jk_5.nailed.server.teamspeak;

import com.typesafe.config.Config;
import jk_5.nailed.server.NailedPlatform;
import jk_5.nailed.server.teamspeak.api.JTS3ServerQuery;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.util.HashMap;

public class TeamspeakManager {

    private static final Config config = NailedPlatform.config().getConfig("teamspeak");
    private static final boolean enabled = config.getBoolean("enabled");
    private static final String host = config.getString("host");
    private static final int port = config.getInt("port");
    private static final String username = config.getString("username");
    private static final String password = config.getString("password");
    private static boolean connected = false;
    private static final Logger logger = LogManager.getLogger();
    private static final JTS3ServerQuery server = new JTS3ServerQuery();

    public static void start(){
        if(!enabled){
            logger.info("Teamspeak integration is disabled in the config");
            return;
        }
        logger.info("Starting teamspeak integration");
        if(!server.connectTS3Query(host, port)){
            displayError();
        }
    }

    private static void displayError(){
        String error = server.getLastError();
        if(error != null){
            //TODO: proper logging
            System.out.println("Teamspeak error:");
            System.out.println(error);
            if(server.getLastErrorPermissionID() != -1) {
                HashMap<String, String> permInfo = server.getPermissionInfo(server.getLastErrorPermissionID());
                if(permInfo != null){
                    System.out.println("Missing Permission: " + permInfo.get("permname"));
                }
            }
        }
    }
}
